src/client/components/modals/Organization.js

```diff
@@ -0,0 +1,251 @@
+    onClick={() => dispatch(requestedCheckMembershipReset(ethereumAddress))}
+  >
+    Back to previous screen
+  </Text.Block>
+)
+
+BackToPreviousScreen.propTypes = {
+  ethereumAddress: PropTypes.string.isRequired,
+  dispatch: PropTypes.func.isRequired,
+}
+
+const CheckInfo = ({
+  type,
+  ethereumAddress,
+  address,
+  dispatch,
+  linkBack = false,
+}) => {
+  const checkInfoIcon = {
+    started: AnimationLoadingCircle,
+    success: IconSuccess,
+    error: IconError,
+  }
+  const checkInfoText = {
+    started: `Validating your membership to ${shortDAOAddress(address)}...`,
+    success: `We found ${shortDAOAddress(
+      address
+    )} and confirmed you are a member`,
+    error: `We could not verify your membership to ${shortDAOAddress(address)}`,
+  }
+
+  return (
+    <ModalWrapper>
+      <CheckWrapper>
+        {checkInfoIcon[type]()}
+        <div
+          css={`
+            height: 100%;
+            width: 100%;
+            position: relative;
+            display: flex;
+            align-items: center;
+          `}
+        >
+          <Text.Block size="xxlarge">{checkInfoText[type]}</Text.Block>
+
+          {linkBack && (
+            <BackToPreviousScreen
+              dispatch={dispatch}
+              ethereumAddress={ethereumAddress}
+            />
+          )}
+        </div>
+      </CheckWrapper>
+    </ModalWrapper>
+  )
+}
+
+CheckInfo.propTypes = {
+  type: PropTypes.string.isRequired,
+  address: PropTypes.string.isRequired,
+  ethereumAddress: PropTypes.string.isRequired,
+  dispatch: PropTypes.func.isRequired,
+  linkBack: PropTypes.bool.isRequired,
+}
+
+CheckInfo.defaultProps = {
+  linkBack: false,
+}
+
+const Organization = ({
+  ethereumAddress,
+  getFormValue,
+  onChange,
+  saveProfile,
+  organizationId,
+  savingError,
+}) => {
+  const [validationErrors, setValidationErrors] = useState({})
+  const { dispatch } = useContext(BoxContext)
+  const { dispatchModal } = useContext(ModalContext)
+  const {
+    checkingMembership,
+    checkedMembershipSuccess,
+    checkedMembershipError,
+    error,
+  } = useProfileStates()
+
+  const checkMembershipError = checkedMembershipError
+    ? { error: `Error checking membership: ${error.message}` }
+    : {}
+
+  const checkMembershipAndSave = async (ethereumAddress, address) => {
+    dispatch(requestedCheckMembership(ethereumAddress))
+
+    try {
+      const result = await fakeIsMember(ethereumAddress, address)
+      if (result) {
+        dispatch(requestedCheckMembershipSuccess(ethereumAddress))
+        saveProfile(ethereumAddress)
+        dispatchModal(close())
+        dispatch(requestedCheckMembershipReset(ethereumAddress))
+        return true
+      } else throw new Error({ message: 'Cannot confirm membership' })
+    } catch (error) {
+      dispatch(requestedCheckMembershipError(ethereumAddress, error))
+      return false
+    }
+  }
+
+  /*
+
+For future reference: ENS resolution can come directly from the Wrapper:
```

Comment:
Hey, let's take these commented lines out and save them in their own iceboxed github ticket. 

```js
if (address.endsWith('.eth')) {
      // address = resolveEnsDomain(address)
      // if (!address) errors['organization'] = 'Could not resolve ENS address'
    } else if (!validateDAOAddress(address))
      errors['organization'] = 'Please provide valid DAO address'

    if (isEmpty(errors)) {
      checkMembershipAndSave(ethereumAddress, address)
    } else {
      setValidationErrors(errors)
    }
  }
```

Please just leave in whatever is necessary for this PR that's being used.

Reply:
Done.